pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -268,65 +275,83 @@ def _should_accumulate(self) -> bool:
         )
         return not accumulation_done and ttp_accumulates_on_final_batch
 
-    def _track_epoch_end_reduce_metrics(
-        self, epoch_output: List[List[STEP_OUTPUT]], batch_end_outputs: STEP_OUTPUT
-    ) -> None:
-        """Adds the batch outputs to the epoch outputs and prepares reduction."""
-        hook_overridden = is_overridden("training_epoch_end", self.trainer.lightning_module)
-        if not hook_overridden:
-            return
-
-        # track the outputs to reduce at the end of the epoch
-        for opt_idx, opt_outputs in enumerate(batch_end_outputs):
-            # with 1 step (no tbptt) don't use a sequence at epoch end
-            if isinstance(opt_outputs, list) and len(opt_outputs) == 1:
-                opt_outputs = opt_outputs[0]
-
-            epoch_output[opt_idx].append(opt_outputs)
-
     @staticmethod
-    def _prepare_outputs(
-        outputs: List[List[List[OutputResult]]], batch_mode: bool
-    ) -> Union[List[List[List[Dict]]], List[List[Dict]], List[Dict], Dict]:
-        """Extract required information from batch or epoch end results.
-
-        Args:
-            outputs: A 3-dimensional list of ``OutputResult`` objects with dimensions:
-                ``[optimizer outs][batch outs][tbptt steps]``.
-
-            batch_mode: If True, ignore the batch output dimension.
+    def _prepare_outputs_training_batch_end(
```

Comment:
@carmocca this code here is now very similar to the one for epoch_end. The only difference is that we have one dimension less and that the final sanitation is different. 

What do you prefer? Merge the functions into one or keep as is?

Reply:
I'd keep them separate, in case the format for a hook changes

Reply:
sounds good